### ISSUE TITLE
`sort -o` should not trigger SC2094

### ISSUE THREAD

**Issue by benhartley** [koalaman/shellcheck#745]
This command:

``` sh
sort -o x < x
```

throws this error:

```
Make sure not to read and write the same file in the same pipeline. [SC2094]
```

This is incorrect since `-o` enables you to [use the same file](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/sort.html):

```
-o  output
    Specify the name of an output file to be used instead of the standard 
    output. This file can be the same as one of the input files.
```

**Comment by brother** [comment]
[SC2094](https://github.com/koalaman/shellcheck/wiki/SC2094) is general purpose not for sort specific. When you have evaluated the use just put a [directive](https://github.com/koalaman/shellcheck/wiki/Directive) before the line and continue on.

#shellcheck disable=SC2094

**Comment by benhartley** [comment]
Fair enough. I just saw [`isHarmlessCommand`](https://github.com/koalaman/shellcheck/blob/6b62b5bf7ecf22529714fed1b08ce603b946f997/ShellCheck/Analytics.hs#L843-L846) and thought it could be a case that gets added to that. I guess it would be hard to account for every single instance of commands where it does not work
